cobald_hep_plugins/stopper.py

```diff
@@ -0,0 +1,65 @@
+from cobald.interfaces import Pool, PoolDecorator
+
+from cobald.utility import enforce
+
+import asyncio
+
+from cobald.daemon import service
+
+
+@service(flavour=asyncio)
+class Stopper(PoolDecorator):
+    """
+    Decorator that sets demand to 0 if the partition has no pending jobs
+
+    :param target: the pool
+    :param script: script that checks for pending jobs on the partition
+    :param interval: interval in seconds between execution of the script
+
+    If there are pending jobs on the partition, the demand is not modified.
+    The demand is set to 0 as long as no pending jobs are detected.
+
+    The default interval is 300 (5 minutes). The script has to be specified.
+    """
+
+    @property
+    def demand(self) -> float:
+        return self._demand
+
+    @demand.setter
+    def demand(self, value: float):
+        self._demand = self._condition_slurm(value)
+        self.target.demand = self._demand
+
+    def _condition_slurm(self, value):
+        """Return 0 if there are no pending jobs, otherwise pass `value`"""
+        if self.n_pend_jobs == 0:
+            return 0
+        else:
+            return value
```

Comment:
You can shorten this to directly evaluating the condition here – it's not used anywhere else, so re-use isn't a concern. Not sure if a ternary `if` or multiple lines is more readable.
```suggestion
        self.target.demand = value if self.n_pend_jobs else 0
```